pkg/block/s3/adapter.go

```diff
@@ -849,6 +849,38 @@ func (a *Adapter) ListParts(ctx context.Context, obj block.ObjectPointer, upload
 	return &partsResp, nil
 }
 
+func (a *Adapter) ListMultipartUploads(ctx context.Context, obj block.ObjectPointer, opts block.ListMultipartUploadsOpts) (*block.ListMultipartUploadsResponse, error) {
+	var err error
+	defer reportMetrics("ListMultipartUploads", time.Now(), nil, &err)
+	bucket, key, qualifiedKey, err := a.extractParamsFromObj(obj)
+	if err != nil {
+		return nil, err
+	}
+	input := &s3.ListMultipartUploadsInput{
+		Bucket:     aws.String(bucket),
+		Prefix:     aws.String(key),
+		MaxUploads: opts.MaxUploads,
```

Comment:
We support MaxUploads, but it means we also need to support KeyMarker in order to perform full listing.
My previous comment gave MaxUploads an as example of one of the capabilities that this API support.
We should verify with product all the functionality we are going to support or not including the fact that this feature will be only supported for s3.